sdk/compute/Microsoft.Azure.Management.Compute/src/Generated/Models/ImageDiskReference.cs

```diff
@@ -31,13 +30,20 @@ public ImageDiskReference()
         /// Initializes a new instance of the ImageDiskReference class.
         /// </summary>
         /// <param name="id">A relative uri containing either a Platform Image
-        /// Repository or user image reference.</param>
+        /// Repository, user image, or Azure Compute Gallery image
+        /// reference.</param>
+        /// <param name="sharedGalleryImageId">A relative uri containing a
+        /// direct shared Azure Compute Gallery image reference.</param>
+        /// <param name="communityGalleryImageId">A relative uri containing a
+        /// community Azure Compute Gallery image reference.</param>
         /// <param name="lun">If the disk is created from an image's data disk,
         /// this is an index that indicates which of the data disks in the
         /// image to use. For OS disks, this field is null.</param>
-        public ImageDiskReference(string id, int? lun = default(int?))
+        public ImageDiskReference(string id = default(string), string sharedGalleryImageId = default(string), string communityGalleryImageId = default(string), int? lun = default(int?))
```

Comment:
same, better to provide backward compatibility

Reply:
please see my comments below, I've added the customizations.